### ISSUE TITLE
Rule Change: `id-denylist` - selective check for object properties etc.

### ISSUE THREAD

**Issue by TokugawaTakeshi** [eslint/eslint#15504]
### What rule do you want to change?

Add options to `id-denylist` allows to deny identifiers selectivery for variables, properties, etc.

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
"id-denylist": [
  "warn",
  {
    "variableDeclarations": [ "data", "err", "lst" ],
    "functionDeclarations": [ "add", "set" ],
    "objectProperties": false
  }
]
```

### What does the rule currently do for this code?

This rules affects to all of 

* variable declarations
* function declarations
* object properties assigned to during object creation
* class fields
* class methods

The variable like `add` can confuse while the `add` as object method coluld be clear (if it is the collection-like object).

Other example. I know that main ESLint package is not responsible for TypeScript, but before `@typescript-eslint` colud adapt the `id-denylist` for enumerations, the selective denying functionality must be implemented in the main package.

```
export enum ValuesTypesIDs {
    number = "NUMBER",
    /* 〔 ESLint muting rationale 〕 This rule is not desired for object keys, but there no option allows to disable it for
     *   the object properties.  */
    /* eslint-disable-next-line id-denylist */
    string = "STRING",
    boolean = "BOOLEAN"
}
```

### What will the rule do after it's changed?

It will become to works selectivelyf for variable names and/or function names and/or properties etc.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules/changes if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
